lib/services/appmanager-service.ts

```diff
@@ -50,14 +52,51 @@ class AppManagerService implements IAppManagerService {
 			let projectName = this.$project.projectData.ProjectName;
 			let solutionPath = buildResult[0].solutionPath;
 			let projectPath = solutionPath.substr(solutionPath.indexOf("/") + 1);
-			this.$server.tam.uploadApplication(projectName, projectName, projectPath).wait();
 
-			this.$logger.info("Successfully uploaded package.");
+			let publishSettings: Server.PublishSettings = {
+				IsPublished: options.publish,
+				NotifyByPush: options.sendPush,
+				NotifyByEmail: options.sendEmail,
+				Groups: []
+			};
+
+			if (options.group.length) {
+				publishSettings.Groups = this.findGroups(options.group).wait();
+			}
 
+			this.$server.tam.uploadApplication(projectName, projectName, projectPath, publishSettings).wait();
+			this.$logger.info("Successfully uploaded package.");
 			this.openAppManagerStore();
 		}).future<void>()();
 	}
 
+	public getGroups(): IFuture<void> {
+		return ((): any  => {
```

Comment:
```TypeScript
return (():  => {
```
is enough. You do not return anything, so any is not appropriate (you can replace it with void or just remove it.